plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -153,16 +150,29 @@ const addToCartWithOptionsStore = store(
 						value,
 					} );
 				}
+
+				const matchedVariation = getMatchedVariation(
+					availableVariations,
+					selectedAttributes
+				);
+				context.variationId = matchedVariation?.variation_id || null;
```

Comment:
So the variation ID is derived from the `availableVariations` and `selectedAttributes`. In this case, I think we should use a derived state for variation ID instead of syncing context.

Imagine that we add a feature to reset the attribute select, if we sync context like we're doing here, we need to sync the context inside the `resetSelection` action. If we derived, we won't have to care about syncing at all. wdyt?

Reply:
If we do so, we need to update the Product Button store to use the derived state instead of context.

Reply:
Yes, good point! Done in d297d68a647bc48dacf8363959ddb28381a1278b.